### DISCUSSION TITLE
Alert ktory wskaze, ktora VMka z backend pool w AppGw nie odpowiada

### DISCUSSION THREAD

**Discussion by Mski89** [AzureCommunityPL/group#52] [Pomoc] [answered]
Cześć, 
Moje AppGw posiada 1 backend pool ktory ma 2 VMki.
Cel jaki chce osiagnac to Redeploy VMki która nie przejdzie Heath Probe. 

Niestety obecny alert "Unhealthy host" w powiadomieniu nie wskazuje na konkreta VMke czy tez IP. 
Moj problem rozwiazywał by alert ktory sprawdza dostepnosc konkretnego adresu www. Ustawił bym to na 2 Vmkach i po problemie. Sęk w tym ze nie ma czegos takiego do wyboru w alertach VM przez co nie do konca wiem jak osiagnac swoj cel.

**Comment by wi5nia** [comment, marked as answer]
Moze np oprzyj sie na tym
https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-diagnostics

**Reply by Mski89** [reply]
Dzieki. 
Wydaje mi sie ze jeszcze VMSS moglo by rozwiazac moj problem.

**Reply by kamil-mrzyglod** [reply]
A próbowałeś używać custom probe'a w AppGW? Powinno dać radę potem wyciągnąć hosta, który nie odpowiada. 

https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-create-probe-portal

**Reply by Mski89** [reply]
@kamil-mrzyglod   tak, mam ustawione. I alert przychodzi ale nie wskazuje na konkretny host. 
Zastanawiam sie czy moge te 2 hosty rozbic na 2 ustawienia HTTP i ustawic 2x custom probe. Wtedy rozroznial bym hosty. Orientujesz sie czy w taki sposob mozna to załatwic ?

**Reply by kamil-mrzyglod** [reply]
Akurat AppGW za bardzo nie używałem więc mam taką bardziej teoretyczną wiedzę. Niemniej jednak tak bym do tego podszedł.

**Reply by Mski89** [reply]
@kamil-mrzyglod 
Get-AzApplicationGatewayBackendHealth listuje stan backendu. Wiec moglbym stad wyciagnac IP ktory nie dziala i dzieki temu uruchamiał by sie skrypt w automation ktory zrobi redeploy VMki. Ma sens wg. Ciebie ?

**Reply by kamil-mrzyglod** [reply]
Jeśli operacja zwraca wynik z wymaganymi wartościami to pewnie, można spróbować :)

**Reply by Mski89** [reply]
Troche rzeźba ale pykło. Dzieki ;)
